well this is alarming, @dwijnand does this alarm you?

```
[info] Set current project to scala-partest (in build file:/home/travis/build/scala/scala-partest/)
Could not create file /home/travis/build/scala/scala-partest/target/streams/$global/projectDescriptors/$global/streams/outjava.io.IOException: No such file or directory
	at sbt.ErrorHandling$.translate(ErrorHandling.scala:10)
	at sbt.IO$.touch(IO.scala:210)
	at sbt.std.Streams$$anon$3$$anon$2.make(Streams.scala:129)
	at sbt.std.Streams$$anon$3$$anon$2.text(Streams.scala:113)
	at sbt.std.Streams$$anon$3$$anon$2.log(Streams.scala:124)
	at sbt.std.TaskStreams$class.log(Streams.scala:56)
	at sbt.std.Streams$$anon$3$$anon$2.log$lzycompute(Streams.scala:102)
	at sbt.std.Streams$$anon$3$$anon$2.log(Streams.scala:102)
	at sbt.Classpaths$$anonfun$depMap$1.apply(Defaults.scala:1613)
	at sbt.Classpaths$$anonfun$depMap$1.apply(Defaults.scala:1612)
	at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:35)
	at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:34)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: No such file or directory
	at java.io.UnixFileSystem.createFileExclusively(Native Method)
	at java.io.File.createNewFile(File.java:1012)
	at sbt.IO$$anonfun$1.apply$mcZ$sp(IO.scala:210)
	at sbt.IO$$anonfun$1.apply(IO.scala:210)
	at sbt.IO$$anonfun$1.apply(IO.scala:210)
	at sbt.ErrorHandling$.translate(ErrorHandling.scala:10)
	at sbt.IO$.touch(IO.scala:210)
	at sbt.std.Streams$$anon$3$$anon$2.make(Streams.scala:129)
	at sbt.std.Streams$$anon$3$$anon$2.text(Streams.scala:113)
	at sbt.std.Streams$$anon$3$$anon$2.log(Streams.scala:124)
	at sbt.std.TaskStreams$class.log(Streams.scala:56)
	at sbt.std.Streams$$anon$3$$anon$2.log$lzycompute(Streams.scala:102)
	at sbt.std.Streams$$anon$3$$anon$2.log(Streams.scala:102)
	at sbt.Classpaths$$anonfun$depMap$1.apply(Defaults.scala:1613)
	at sbt.Classpaths$$anonfun$depMap$1.apply(Defaults.scala:1612)
	at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:35)
	at scala.Function4$$anonfun$tupled$1.apply(Function4.scala:34)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] (*:projectDescriptors) Could not create file /home/travis/build/scala/scala-partest/target/streams/$global/projectDescriptors/$global/streams/outjava.io.IOException: No such file or directory
```

--- a/src/main/scala/scala/tools/partest/PartestTask.scala
+++ b/src/main/scala/scala/tools/partest/PartestTask.scala
@@ -11,9 +11,8 @@ package partest
 
 import scala.tools.nsc.Properties.propOrFalse
 import scala.tools.ant.sabbus.CompilationPathProperty
-import java.lang.reflect.Method
 import org.apache.tools.ant.Task
-import org.apache.tools.ant.types.{ Reference, FileSet }
+import org.apache.tools.ant.types.Reference
 import org.apache.tools.ant.types.Commandline.Argument
 import scala.tools.ant.ScalaTask
 import nest.NestUI

--- a/src/main/scala/scala/tools/partest/StoreReporterDirectTest.scala
+++ b/src/main/scala/scala/tools/partest/StoreReporterDirectTest.scala
@@ -2,7 +2,6 @@ package scala.tools.partest
 
 import scala.tools.nsc.Settings
 import scala.tools.nsc.reporters.StoreReporter
-import scala.collection.mutable
 
 trait StoreReporterDirectTest extends DirectTest {
   lazy val storeReporter: StoreReporter = new scala.tools.nsc.reporters.StoreReporter()

--- a/src/main/scala/scala/tools/partest/TestState.scala
+++ b/src/main/scala/scala/tools/partest/TestState.scala
@@ -1,7 +1,5 @@
 package scala.tools.partest
 
-import scala.tools.nsc.util.stackTraceString
-
 sealed abstract class TestState {
   def testFile: java.io.File
   def what: String

--- a/src/main/scala/scala/tools/partest/nest/AbstractRunner.scala
+++ b/src/main/scala/scala/tools/partest/nest/AbstractRunner.scala
@@ -9,7 +9,7 @@ package nest
 
 import utils.Properties._
 import scala.tools.nsc.Properties.{ versionMsg, propOrFalse, setProp }
-import scala.collection.{ mutable, immutable }
+import scala.collection.mutable
 import TestKinds._
 import scala.reflect.internal.util.Collections.distinctBy
 

--- a/src/main/scala/scala/tools/partest/nest/DirectCompiler.scala
+++ b/src/main/scala/scala/tools/partest/nest/DirectCompiler.scala
@@ -7,12 +7,10 @@ package scala.tools.partest
 package nest
 
 import scala.collection.mutable.ListBuffer
-import scala.tools.nsc.{ Global, Settings, CompilerCommand, FatalError }
+import scala.tools.nsc.{ Global, Settings, CompilerCommand }
 import scala.tools.nsc.reporters.{ Reporter, ConsoleReporter }
-import scala.tools.nsc.util.{ FakePos, stackTraceString }
 import scala.reflect.io.AbstractFile
-import scala.reflect.internal.util.Position
-import java.io.{ BufferedReader, PrintWriter, FileReader, Writer, FileWriter }
+import java.io.{ PrintWriter, FileWriter }
 
 class ExtConsoleReporter(settings: Settings, val writer: PrintWriter) extends ConsoleReporter(settings, Console.in, writer) {
   shortname = true
@@ -72,7 +70,6 @@ class DirectCompiler(val runner: Runner) {
 
   def compile(opts0: List[String], sources: List[File]): TestState = {
     import runner.{ sources => _, _ }
-    import ClassPath.{join, split}
 
     // adding codelib.jar to the classpath
     // codelib provides the possibility to override standard reify

--- a/src/main/scala/scala/tools/partest/nest/FileManager.scala
+++ b/src/main/scala/scala/tools/partest/nest/FileManager.scala
@@ -8,16 +8,7 @@
 package scala.tools.partest
 package nest
 
-import java.io.{
-  File,
-  IOException,
-  OutputStreamWriter,
-  FileOutputStream
-}
-import java.net.URI
-import scala.reflect.io.AbstractFile
-import scala.collection.mutable
-import scala.reflect.internal.util.ScalaClassLoader
+import java.io.{ File, IOException }
 import java.net.URLClassLoader
 
 object FileManager {

--- a/src/main/scala/scala/tools/partest/nest/PathSettings.scala
+++ b/src/main/scala/scala/tools/partest/nest/PathSettings.scala
@@ -5,7 +5,6 @@
 package scala.tools.partest
 package nest
 
-import scala.tools.nsc.util.ClassPath
 import scala.tools.nsc.io.{ Path, File, Directory }
 import scala.tools.nsc.Properties.{ propOrNone }
 import Path._
@@ -41,19 +40,6 @@ object PathSettings {
 
   // Directory <root>/test/files or .../scaladoc
   def srcDir = Directory(testRoot / testSourcePath toCanonical)
-
-  // Directory <root>/test/files/lib
-  private def srcLibDir = Directory(srcDir / "lib")
-
-  // Directory <root>/build
-  private def buildDir: Directory = {
-    val bases      = testRoot :: testRoot.parents
-    // In the classic "ant" build, the relevant subdirectory is called build,
-    // but in the postmodern "sbt" build, it is called target.  Look for both.
-    val dirs = Path.onlyDirs(bases flatMap (x => List(x / "build", x / "target")))
-
-    dirs.headOption getOrElse sys.error("Neither 'build' nor 'target' dir found under test root " + testRoot + ".")
-  }
 
   def srcSpecLib     = findJar("instrumented", Directory(srcDir / "speclib"))
   def srcCodeLib     = findJar("code",  Directory(srcDir / "codelib"), Directory(testRoot / "files" / "codelib") /* work with --srcpath pending */)

--- a/src/main/scala/scala/tools/partest/nest/Runner.scala
+++ b/src/main/scala/scala/tools/partest/nest/Runner.scala
@@ -6,25 +6,22 @@ package scala.tools.partest
 package nest
 
 import java.io.{ Console => _, _ }
-import java.net.URL
-import java.nio.charset.{ Charset, CharsetDecoder, CharsetEncoder, CharacterCodingException, CodingErrorAction => Action }
 import java.util.concurrent.Executors
 import java.util.concurrent.TimeUnit.NANOSECONDS
 import scala.collection.mutable.ListBuffer
 import scala.concurrent.duration.Duration
-import scala.io.Codec
 import scala.reflect.internal.FatalError
 import scala.reflect.internal.util.ScalaClassLoader
 import scala.sys.process.{ Process, ProcessLogger }
-import scala.tools.nsc.Properties.{ envOrNone, isWin, jdkHome, javaHome, propOrEmpty, setProp, versionMsg, javaVmName, javaVmVersion, javaVmInfo }
+import scala.tools.nsc.Properties.{ envOrNone, isWin, javaHome, propOrEmpty, versionMsg, javaVmName, javaVmVersion, javaVmInfo }
 import scala.tools.nsc.{ Settings, CompilerCommand, Global }
 import scala.tools.nsc.reporters.ConsoleReporter
-import scala.tools.nsc.util.{ Exceptional, stackTraceString }
+import scala.tools.nsc.util.stackTraceString
 import scala.util.{ Try, Success, Failure }
 import ClassPath.{ join, split }
 import TestState.{ Pass, Fail, Crash, Uninitialized, Updated }
 
-import FileManager.{ compareFiles, compareContents, joinPaths, withTempFile }
+import FileManager.{ compareContents, joinPaths, withTempFile }
 
 trait TestInfo {
   /** pos/t1234 */

--- a/src/main/scala/scala/tools/partest/package.scala
+++ b/src/main/scala/scala/tools/partest/package.scala
@@ -7,7 +7,7 @@ package scala.tools
 import java.util.concurrent.{ Callable, ExecutorService }
 import scala.concurrent.duration.Duration
 import scala.sys.process.javaVmArguments
-import scala.tools.nsc.util.{ ScalaClassLoader, Exceptional }
+import scala.tools.nsc.util.Exceptional
 
 package object partest {
   type File         = java.io.File

